File: python2/tests/subgrgens.py
import sys
sys.path.insert(1, sys.path[0] + '/..')
import groups
from groups.about import about

g = groups.Symmetric(4)
data = about(g)
for subgr in sorted(data["subgroups"], key=lambda s: data["subgroups"][s]["index"]):
    print '{' + ', '.join(data["subgroups"][subgr]["elements"]) + '}'
    for gen in data["subgroups"][subgr]["generators"]:
	print ' {' + ', '.join(gen) + '}'
    print

File: haskell/tests/subgrgens.hs
import Data.List (intercalate)
import qualified Data.Map as Map
import qualified Data.Set as Set
import Groups (Group, symmetric)
import Groups.Subgroups (subgroupGens)
import Permutation (Permutation, showCycles)

gr :: Group Permutation
gr = symmetric 4

showElem :: Permutation -> String
showElem = showCycles

showElemSet :: Set.Set Permutation -> String
showElemSet xs = '{' : intercalate ", " (map showElem $ Set.toList xs) ++ "}"

main :: IO ()
main = mapM_ (\(subgr, gens) -> do putStrLn $ showElemSet subgr
				   mapM_ (\g -> putStrLn $ ' ' : showElemSet g)
				    $ Set.toList gens
				   putChar '\n') $ Map.toList $ subgroupGens gr
